src/roktManager.ts

```diff
@@ -167,27 +171,49 @@ export default class RoktManager {
             const { attributes } = options;
             const sandboxValue = attributes?.sandbox || null;
             const mappedAttributes = this.mapPlacementAttributes(attributes, this.placementAttributesMapping);
+            const { hashedEmailUserIdentityType = null } = this.configSettings || {};
+
+            const normalizedHashedEmailUserIdentityType = hashedEmailUserIdentityType?.toLowerCase() || null;
 
             // Get current user identities
             this.currentUser = this.identityService.getCurrentUser();
             const currentUserIdentities = this.currentUser?.getUserIdentities()?.userIdentities || {};
+
             const currentEmail = currentUserIdentities.email;
             const newEmail = mappedAttributes.email as string;
 
+            let currentHashedEmail: string | undefined;
+            let newHashedEmail: string | undefined;
+            
+            // Hashed email identity is valid if it is set to Other-Other10
+            if(normalizedHashedEmailUserIdentityType && IdentityType.getIdentityType(normalizedHashedEmailUserIdentityType) !== false) {
+                currentHashedEmail = currentUserIdentities[normalizedHashedEmailUserIdentityType];
+                newHashedEmail = mappedAttributes['emailsha256'] as string || undefined;
+            }
+
+            const emailChanged = !!(newEmail && (!currentEmail || currentEmail !== newEmail));
```

Comment:
So !! is a double not right so it doesn't invert like not(!), it just converts to a boolean?

So this would be = ((newEmail != nil &&  currentEmail == nil) || (newEmail != nil &&  currentEmail !== newEmail))